contracts/AlephZero_SpokePool.sol

```diff
@@ -0,0 +1,135 @@
+// SPDX-License-Identifier: BUSL-1.1
+
+// Arbitrum only supports v0.8.19
+// See https://docs.arbitrum.io/for-devs/concepts/differences-between-arbitrum-ethereum/solidity-support#differences-from-solidity-on-ethereum
+pragma solidity ^0.8.19;
+
+import "./SpokePool.sol";
+import "./libraries/CircleCCTPAdapter.sol";
+
+interface StandardBridgeLike {
+    function outboundTransfer(
+        address _l1Token,
+        address _to,
+        uint256 _amount,
+        bytes calldata _data
+    ) external payable returns (bytes memory);
+}
+
+/**
+ * @notice AVM specific SpokePool. Uses AVM cross-domain-enabled logic to implement admin only access to functions.
+ * @custom:security-contact bugs@across.to
+ */
+contract AlephZero_SpokePool is SpokePool, CircleCCTPAdapter {
+    // Address of the Arbitrum L2 token gateway to send funds to L1.
+    address public l2GatewayRouter;
+
+    // Admin controlled mapping of arbitrum tokens to L1 counterpart. L1 counterpart addresses
+    // are necessary params used when bridging tokens to L1.
+    mapping(address => address) public whitelistedTokens;
+
+    event SetL2GatewayRouter(address indexed newL2GatewayRouter);
+    event WhitelistedTokens(address indexed l2Token, address indexed l1Token);
+
+    /// @custom:oz-upgrades-unsafe-allow constructor
+    constructor(
+        address _wrappedNativeTokenAddress,
+        uint32 _depositQuoteTimeBuffer,
+        uint32 _fillDeadlineBuffer,
+        IERC20 _l2Usdc,
+        ITokenMessenger _cctpTokenMessenger
+    )
+        SpokePool(_wrappedNativeTokenAddress, _depositQuoteTimeBuffer, _fillDeadlineBuffer)
+        CircleCCTPAdapter(_l2Usdc, _cctpTokenMessenger, CircleDomainIds.Ethereum)
```

Comment:
Is it ok to set the Circle Domain ID to ethereum here? Or should it be 'UNINITIALIZED'? cc @bmzig

Reply:
Either way is fine. I think we should keep it for consistency, since we have this done for the other spoke pool which don't have CCTP support (like the [Ovm_SpokePools](https://github.com/across-protocol/contracts/blob/3cfd778c1bd6d805b10cbe91f794bb24b060c26a/contracts/Ovm_SpokePool.sol#L75C18-L75C19)). You are correct in that the main thing which needs to be zero is the token messenger.

Reply:
Perfect, thank u Bennet!